parser-typechecker/src/Unison/Typechecker/Context.hs

```diff
@@ -1249,6 +1249,10 @@ synthesizeWanted e
           et = existential' l B.Blank e
       appendContext $
         [existential i, existential e, existential o, Ann arg it]
+      when (Var.typeOf i == Var.Delay) $ do
+        -- '(1 + 1) turns into a lambda with an arg variable of type Var.Delay
+        -- here's where the typechecker assumes this must be of type 'thunkArgType'
+        subtype it (DDB.thunkArgType l)
```

Comment:
Previously, we were ending up with inferred thunks that had types like `forall a . a -> Blah` since they didn't use their argument. Which is actually inconsistent, since if you write out the signature of the thunk, it will get a type of `Unit -> ...`. This causes some round trip issues (you'd infer the thunk, but then when pretty-printing it out, it'd get a "different" type because the signature had been added).